micro/goal.go

```diff
@@ -26,25 +27,21 @@ func EmptyState() *State {
 }
 
 // Substitutions is a list of substitutions represented by a sexprs pair.
-type Substitutions map[string]*ast.SExpr
+type Substitutions map[uint64]*ast.SExpr
 
 func (s Substitutions) String() string {
 	sexprs := make([]*ast.SExpr, len(s))
-	ss := map[string]*ast.SExpr(s)
+	ss := map[uint64]*ast.SExpr(s)
 	for i, k := range deriveSorted(deriveKeys(ss)) {
 		v := s[k]
-		sexprs[len(s)-1-i] = ast.Cons(&ast.SExpr{Atom: &ast.Atom{Var: &ast.Variable{Name: k}}}, v)
+		vv := ast.NewVar(fmt.Sprintf("v%d", k), k)
```

Comment:
Isn't this just micro.Var instead of ast.NewVar

Reply:
Yes you're right. I'll change that to take a uint64 while I'm at it